### DISCUSSION TITLE
Send message to non-child/parent actors

### DISCUSSION THREAD

**Discussion by saltzm** [statelyai/xstate#1582] [Q&A]
Hi! It seems not possible to send a message from one actor to another one that is not a child/parent. It's not documented and it's forbidden in code [here](https://github.com/davidkpiano/xstate/blob/ebd91ce7ac377e372cf56ec9186039dd5429e991/packages/core/src/interpreter.ts#L660-L663).

Imagine A is the parent of B and C, and B needs to send a message to C. Is that possible?

Is sending a message to a non-child/parent actor an antipattern? The workaround I can think of is to send a message back up to the common parent, and then have it forward it back down - in this case B would send a message to A, and then A would redirect it to C.

**Comment by Andarist** [comment]
You can fairly easy implement address sharing to enable direct communication:
https://codesandbox.io/s/dank-fog-tr4yf?file=/src/index.js

**Comment by saltzm** [comment]
Interesting, thanks for the example. I was doing something wrong and moved too quickly before posting :) Going to keep playing with my own example and see if I can get it working.
